modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/is_object.ts

```diff
@@ -16,7 +16,7 @@
  * Returns true if the specified value is an object. This includes arrays
  * and functions.
  */
-export default function isObject(val: any) {
+export default function isObject<T>(val: T): val is NonNullable<T> {
```

Comment:
I think this should infer `never` when called with other types. For example, `const a = 'a'; if (isObject('a')) {...}` will consider that a is of type `string` instead of `never`.